### PR TITLE
feat: add lnurl withdrawal

### LINKED ISSUE #6: [Feature Request] LNURL-withdraw

### PR DESCRIPTION
**Pull request by im-adithya**
Fixes #6 

## Description
Adds LNURL Withdraw

## Screenshots
<table>
<tr>
<td>

</td>
<td>

</td>
<td>

</td>

<tr>
<td>

</td>
<td>

</td>
<td>

</td>
</tr>

<tr>
<td>

</td>
<td>

</td>
<td>

</td>
</tr>
</table>

### CODE REVIEW

**Comment by rolznz** [comment]
@im-adithya nice! what did you use for generating the two different types of vouchers?

**Comment by im-adithya** [comment]
I'm using the LNURLw plugin from LNBits here (ran an instance for 21 sats, funded the wallet with NWC and installed the plugin and followed [this video](https://www.youtube.com/watch?v=TUmsHpJtveQ&ab_channel=lnbits) (make sure you install v0.1.13 otherwise it won't work))

**Review by reneaaron — CHANGES_REQUESTED** [review]
Nice, just did some first testing. :muscle: 

1) Min & Max amounts
If it's a dynamic amount lnurl-w (min != max) the UX currently is a bit flawed. I don't know what the min / max values are and the button is just disabled. 

What if we do the on "Confirm" and display a toast with the min / max values instead? Since this also needs to be applied to the lnurl-pay flow I guess we can also do it as a follow-up: 

https://github.com/getAlby/go/issues/159

2) Linking
You already added a todo, please make sure this works as it's quite popular for different services (such as lightsats) to use the lightning:LNURLW link that just takes you to the wallet

**Comment by im-adithya** [comment]
Addressed ✅ 

> Min & Max amounts

For now I added code to throw error if amount is < min or > max, confirm button won't be disabled

File: lib/link.ts [hunk 1]
```diff
@@ -37,6 +38,20 @@ export const handleLink = (url: string) => {
...
+      if (lnurlDetails.tag === "withdrawRequest") {
+        router.push({
+          pathname: "/withdraw",
+          params: {
+            url: fullUrl,
+          },
+        });
+      }
+    }
+
```
Comment (im-adithya) [inline lib/link.ts:54]: Could've added a check for "payRequest" but we don't strip `lightning:` here, should we make `getDetails` handle it?

Comment (im-adithya) [reply lib/link.ts:54]: Sorry I mean make `findLnurl` return a lightning address as well, currently it doesn't return `hello@getalby.com` when `lightning:hello@getalby.com` is passed

File: hooks/__tests__/useHandleLinking.ts [hunk 2]
```diff
@@ -3,58 +3,67 @@ import { router } from "expo-router";
...
-  "lightning://hello@getalby.com": "lightning:hello@getalby.com",
-  "LIGHTNING://hello@getalby.com": "lightning:hello@getalby.com",
-  "LIGHTNING:hello@getalby.com": "lightning:hello@getalby.com",
-  "lightning:lnbc1": "lightning:lnbc1",
-  "lightning://lnbc1": "lightning:lnbc1",
-  "bitcoin:bitcoinaddress?lightning=invoice":
-    "bitcoin:bitcoinaddress?lightning=invoice",
-  "BITCOIN:bitcoinaddress?lightning=invoice":
-    "bitcoin:bitcoinaddress?lightning=invoice",
+const lnurlWithdrawLink = "" // "lnurl1dp68gurn8ghj7mrfva58gumpw3ejucm0d5hkzurf9amkjargv3exzampd3xxjmntwvhkvepcvcckvc3s94jnqerz956r2wtr95urqcfc95urqcnpxd3rqc3evv6kx5kckkv"
```
[the remark this answers is not in the document]
Comment (im-adithya) [reply hooks/__tests__/useHandleLinking.ts]: Done ✅

File: lib/link.ts (hunk 1 above)
Comment (reneaaron) [reply lib/link.ts:55]: Yeah, but let's do it as a follow-up. Let's also add some tests for this lib function.

Comment (reneaaron) [reply lib/link.ts:55]: https://github.com/getAlby/go/issues/162

File: pages/withdraw/Withdraw.tsx [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+                      <Text className="text-muted-foreground text-center font-semibold2">
+                        Description
+                      </Text>
+                      <Text className="text-center text-foreground text-2xl font-medium2">
+                        {lnurlDetails.defaultDescription}
+                      </Text>
+                    </View>
+                  </View>
+                  <View className="m-6">
+                    <Button
```
Comment (reneaaron) [inline pages/withdraw/Withdraw.tsx:247]: The same button is defined twice here? Can we make it so it's only one?

File: hooks/__tests__/useHandleLinking.ts [hunk 4]
```diff
@@ -3,58 +3,113 @@ import { router } from "expo-router";
...
+    url: "bitcoin:bitcoinaddress?lightning=invoice",
+    path: "/send",
+  },
+  "BITCOIN:bitcoinaddress?lightning=invoice": {
+    url: "bitcoin:bitcoinaddress?lightning=invoice",
+    path: "/send",
+  },
+
+  // LNURL-withdraw
+  "lightning:lnurlw123": { url: "lightning:lnurlw123", path: "/withdraw" },
```
Comment (im-adithya) [inline hooks/__tests__/useHandleLinking.ts:67]: Ah nice, thanks @reneaaron 🙌

File: pages/withdraw/Withdraw.tsx (hunk 3 above)
Comment (im-adithya) [reply pages/withdraw/Withdraw.tsx:247]: Yup, changing

**Review by reneaaron — APPROVED** [review]
tACK

**Pull request merged**